spec/compiler/parser/parser_spec.cr

```diff
@@ -1508,6 +1508,12 @@ module Crystal
       it_parses "{% begin %}%q#{open} %s #{close}{% end %}", MacroIf.new(true.bool, MacroLiteral.new("%q#{open} %s #{close}"))
     end
 
+    it_parses %(foo(bar:"a", baz:"b")), Call.new(nil, "foo", named_args: [NamedArgument.new("bar", "a".string), NamedArgument.new("baz", "b".string)])
+    it_parses %(foo(bar:a, baz:b)), Call.new(nil, "foo", named_args: [NamedArgument.new("bar", "a".call), NamedArgument.new("baz", "b".call)])
+    it_parses %({foo:"a", bar:"b"}), NamedTupleLiteral.new([NamedTupleLiteral::Entry.new("foo", "a".string), NamedTupleLiteral::Entry.new("bar", "b".string)])
+    it_parses %({foo:'a', bar:'b'}), NamedTupleLiteral.new([NamedTupleLiteral::Entry.new("foo", CharLiteral.new('a')), NamedTupleLiteral::Entry.new("bar", CharLiteral.new('b'))])
+    it_parses %({foo:a, bar:b}), NamedTupleLiteral.new([NamedTupleLiteral::Entry.new("foo", "a".call), NamedTupleLiteral::Entry.new("bar", "b".call)])
```

Comment:
What happens with `{foo::bar, baz::qux}` ?

Reply:
Good question. It gives a syntax error. I think it's fine, I wouldn't parse that as `{foo: :bar, baz: :qux}`.

Reply:
mainly because `::` has a different meaning

Reply:
`foo ::bar` is `foo(::bar)`
But `::` as a namespace separator is only that between `Constants`

I'm fine either way :-)

Reply:
Yes, in Crystal. In Ruby you can do that with any expression, or at least with var. I wouldn't discard this in the future.